src/main/java/uk/gov/ons/ctp/response/casesvc/message/PubSubSubscription.java

```diff
@@ -0,0 +1,63 @@
+package uk.gov.ons.ctp.response.casesvc.message;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.ctp.response.casesvc.config.AppConfig;
+import uk.gov.ons.ctp.response.casesvc.message.feedback.CaseReceipt;
+import uk.gov.ons.ctp.response.casesvc.message.feedback.InboundChannel;
+import uk.gov.ons.ctp.response.lib.common.error.CTPException;
+
+@Component
+public class PubSubSubscription {
+    private static final Logger log = LoggerFactory.getLogger(PubSubSubscription.class);
+    @Autowired private AppConfig appConfig;
+    @Autowired private CaseReceiptReceiver caseReceiptReceiver;
+
+    @EventListener(ApplicationReadyEvent.class)
+    public void caseNotificationSubscription() {
+        ProjectSubscriptionName subscriptionName =
+                ProjectSubscriptionName.of(
+                        appConfig.getGcp().getProject(),
+                        appConfig.getGcp().getReceiptSubscription());
+        MessageReceiver receiver = createMessageReceiver();
+        Subscriber subscriber;
+        subscriber = Subscriber.newBuilder(subscriptionName, receiver).build();
+        subscriber.startAsync().awaitRunning();
+        log.with("subscription", subscriptionName.toString()).info("Listening for messages on subscription");
+    }
+
+    public MessageReceiver createMessageReceiver() {
+        return (PubsubMessage message, AckReplyConsumer consumer) -> {
+            String payload = message.getData().toStringUtf8();
+            log.with("payload", payload).info("Received a receipt");
+            ObjectMapper mapper = new ObjectMapper();
+            CaseReceipt receipt;
```

Comment:
i think this can live inside the try scope

Reply:
Agreed.  Done